bumpversion/__init__.py

```diff
@@ -336,7 +342,59 @@ def __iter__(self):
         return iter(self._values)
 
     def __repr__(self):
-        return '<bumpversion.Version:{}>'.format(keyvaluestring(self._values))
+        by_part = ", ".join("{}={}".format(k, v) for k, v in self.items())
+        return '<bumpversion.Version:{}>'.format(by_part)
+
+    def __hash__(self):
+        return hash(tuple((k, v) for k, v in self.items()))
+
+    def _compare(self, other, method, strict=True):
```

Comment:
This function is really hard to understand at first sight. The `strict` parameter should be named something clearer, perhaps `allow_equal`?

Reply:
Indeed, I have changed the logic and improved the docs so the intent is more clear